tests/unit/checkmate/checker/url/url_haus_test.py

```diff
@@ -0,0 +1,89 @@
+from unittest.mock import sentinel
+
+import pytest
+from h_matchers import Any
+from httpretty import httprettified, httpretty
+from pkg_resources import resource_string
+
+from checkmate.checker.url import URLHaus
+from checkmate.checker.url.reason import Reason
+
+
+class TestURLHaus:
+    def test_check_url_with_hits(self, URLHausRule):
+        URLHausRule.find_matches.return_value.count.return_value = 2
+
+        response = URLHaus(sentinel.db_session).check_url(["hex_hash"])
+
+        assert response == Any.generator.containing([Reason.MALICIOUS]).only()
+        # The above will exchaust the generator and allow us to make assertions
+        # about the call
+        URLHausRule.find_matches.assert_called_once_with(
+            sentinel.db_session, ["hex_hash"], limit=1
+        )
+
+    def test_check_url_with_no_hits(self, URLHausRule):
+        URLHausRule.find_matches.return_value.count.return_value = 0
+
+        response = URLHaus(sentinel.db_session).check_url(["hex_hash"])
+
+        assert response == Any.generator.containing([]).only()
+
+    @httprettified
+    def test_reinitialise_db(self, URLHausRule):
```

Comment:
We just kind of run the pipeline for real here. I think this is easier and better than trying to mock out the stages or anything. It also functions as a mini integration test, to ensure you don't have stages in a goofy order or something.